packages/blockchain-extension/extension/commands/packageSmartContractCommand.ts

```diff
@@ -216,15 +218,65 @@ async function packageJsonNameAndVersion(workspaceDir: vscode.WorkspaceFolder, o
+
+    const contractSemver: semver.SemVer = semver.coerce(contractAPI); // Contract semver
+
+    const contractMajor: number = contractSemver.major;
+
+    // Check if fabric contract API version is greater than 2.0.0
+    if (!(contractMajor >= 2)) {
+        // Version is not greater than or equal to 2
+        throw new Error(`In this version of the extension, you can only package contracts which use fabric-contract-api v2.0.0 or greater. Your version: ${contractAPI}`);
+    }
+
     return { workspacePackageName, workspacePackageVersion };
 }
 
 /**
  * Method which calls an input box should the project be coded in java, which asks the user for a package name and version
  * (as java projects do not contain a package.json file), and returns an object containing both these values.
+ * @param workspaceDir {String} workspaceDir A string containing the path to the current active workspace (the workspace of the project the user is packaging).
  * @returns {string, string} Returns an object with the workspacePackageName and workspacePackageVersion which will be used in the createPackageDir() method
  */
-async function javaPackageAndVersion(overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+async function javaPackageAndVersion(workspaceDir: vscode.WorkspaceFolder, overrideName?: string, overrideVersion?: string): Promise<{ workspacePackageName: string, workspacePackageVersion: string }> {
+
+    let contractVersion: string = 'fabric-chaincode-shim';
+    let workspaceGradleMaven: string = path.join(workspaceDir.uri.fsPath, '/build.gradle');
+    if (!(await fs.pathExists(workspaceGradleMaven))) {
+        workspaceGradleMaven = path.join(workspaceDir.uri.fsPath, 'pom.xml');
+        contractVersion = 'fabric-chaincode-java';
+    }
+
+    const workspaceGradleContents: string = await fs.readFile(workspaceGradleMaven, 'utf8');
+
+    const regex: RegExp = new RegExp(/fabric-chaincode-shim:(.*)|'fabric-chaincode-shim', version: '(.*)'|fabric-chaincode-java.version>(.*)</);
+
+    const result: RegExpExecArray = regex.exec(workspaceGradleContents);
+
+    if (!result) {
+        const warnPackage: boolean = await UserInputUtil.showConfirmationWarningMessage(`Could not find the ${contractVersion} version. Would you like to package it anyway?`);
+        if (!warnPackage) {
+            return;
+        }
+    } else {
+
+        let contractSemver: semver.SemVer;
+        let version: string;
+        for (let i: number = 1; i < result.length; i++) {
+            if (result[i]) {
+                contractSemver = semver.coerce(result[i]); // Contract semver
+                version = result[i];
+            }
+        }
+
+        const contractMajor: number = contractSemver.major;
+        if (!(contractMajor >= 2)) {
+            throw new Error(`In this version of the extension, you can only package contracts which use fabric v2 or greater. Your version: ${version}`);
```

Comment:
`Unable to package contract. Contract API dependency must support Fabric 2` - we should make all error messages consistent and something like this.